test/integration/db/db_tests.dart

```diff
@@ -674,31 +674,28 @@ Future waitUntilEntitiesGone(db.DatastoreDB mdb,
 Future waitUntilEntitiesHelper(db.DatastoreDB mdb,
                                List<db.Key> keys,
                                bool positive,
-                               db.Partition partition) {
+                               db.Partition partition) async {
   var keysByKind = <Type, List<db.Key>>{};
   for (var key in keys) {
     keysByKind.putIfAbsent(key.type, () => <db.Key>[]).add(key);
   }
 
-  Future waitForKeys(Type kind, List<db.Key> keys) {
-    return mdb.query(kind, partition: partition)
-        .run().toList().then((List<db.Model> models) {
+  Future waitForKeys<T extends db.Model>(List<db.Key> keys) {
+    return mdb.lookup<T>(keys).then((List<T> models) {
```

Comment:
OK.  I can do this, but it is going to illustrate why having type parameters for the Key class is a really good idea.  The boilerplate required to make this work is pretty ugly.

Reply:
It's unfortunate, but I don't think having the keys use a `Type` instead of a `String` changes the problem here.

Previously the code would first build up a map from kind to keys and then issue one query per kind. This worked because the kind was represented as `Type` and could be passed **as a value argument** to `query` (it was the first parameter).

The breaking change made this impossible now, because, even though we could build the same `keysByKind` map and represent the kind by `Type`, we would **no longer be able** to issue queries based on a kind we get from `keysByKind.forEach((Type kind, ...) { ... }`, since it's not possible to do `Type kind;  await (mdb.query<kind>(...).run())`.

Now we have to pass the kind **as a type argument** to `query<>()`.

Reply:
Yes, on further reflection working around that limitation is no better than what you suggested in the first place as we'd have to pass in multiple type arguments and use whereType or other such shenanigans.

In any event, implemented as you suggested and things seem to work now.  Thanks for the assist in unblocking this.